#! /usr/bin/env python
#*  All rights reserved.
#*
#*  Redistribution and use in source and binary forms, with or without
#*  modification, are permitted provided that the following conditions
#*  are met:
#*
#*   * Redistributions of source code must retain the above copyright
#*     notice, this list of conditions and the following disclaimer.
#*   * Redistributions in binary form must reproduce the above
#*     copyright notice, this list of conditions and the following
#*     disclaimer in the documentation and/or other materials provided
#*     with the distribution.
#*   * Neither the name of the Willow Garage nor the names of its
#*     contributors may be used to endorse or promote products derived
#*     from this software without specific prior written permission.
#*
#*  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
#*  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
#*  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
#*  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
#*  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
#*  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
#*  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
#*  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
#*  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
#*  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
#*  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
#*  POSSIBILITY OF SUCH DAMAGE.
#***********************************************************

# Author: Blaise Gassend

# Forearm camera configuration

PACKAGE='multi_interface_roam'
import roslib; roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator import *

gen = ParameterGenerator()

band_2400_MHz = gen.const("2400_MHz", int_t, 1, "Use only the 2.4 GHz band.")
band_5000_MHz = gen.const("5000_MHz", int_t, 2, "Use only the 5 GHz band.")
band_both = gen.const("Both", int_t, 3, "Use both the 2.4 GHz and 5 GHz bands.")

band_enum = gen.enum([band_2400_MHz, band_5000_MHz, band_both], "The band to operate on.")

#       Name                    Type      Reconfiguration level             Description                                                                                      Default    Min   Max
gen.add("interface",            str_t,    0,                                "Non-empty to force the use of a particular interface.",                                         "")
gen.add("scan_only",            bool_t,   0,                                "Indicates that no association attempts should be made.",                                        False)
gen.add("use_tunnel",           bool_t,   0,                                "Indicates whether the default route should be via the tunnel interface.",                       True)
gen.add("disable_ping_timeout", bool_t,   0,                                "Disables timing out when ping packets are not being received.",                                 False)
gen.add("bssid",                str_t,    0,                                "Non-empty to force association to a particular bssid.",                                         "")
gen.add("ssid",                 str_t,    0,                                "Non-empty to force association to a particular ssid.",                                          "")
gen.add("band",                 int_t,    0,                                "Non-empty to force association to a particular ssid.",                                          2, 1, 3, edit_method = band_enum)

exit(gen.generate(PACKAGE, "multi_interface_roam", "MultiInterfaceRoam"))
